src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -701,6 +681,21 @@ internal override LocalSymbol GetAdjustedLocalSymbol(LocalSymbol local, int posi
 
         private LocalFunctionSymbol GetDeclaredLocalFunction(LocalFunctionStatementSyntax declarationSyntax, SyntaxToken declaredIdentifier)
         {
+            // If nullable semantic analysis is enabled, we need to potentially reinfer the containing symbol
+            // of a local function, as it could be in a reinferred lambda. For an example, see
+            // GetSymbolInfo_NestedLambdaReinference_LocalFunctionInLambda
+            if (Compilation.NullableSemanticAnalysisEnabled)
+            {
+                GetBoundNodes(declarationSyntax, out _, out var lowestNode, out _, out _);
+
+                if (lowestNode is BoundLocalFunctionStatement localFunctionNode)
+                {
+                    return localFunctionNode.Symbol;
+                }
+
+                return null;
```

Comment:
>return null; [](start = 16, length = 12)

Should we just fall through to the default handling? It looks like this is what would happen for local symbol.